File: CS Practices/Example of Data Entry/C#/PhoneBook.Vb/FindPerson.vb
' Import the framework of the Libraries
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports System.Data.Entity

' PhoneBook project
Namespace PhoneBook
    ' class FindPerson
    Class FindPerson
        ' private field of type PhooneBookContext
        Private Property ctxFind() As PhoneBookContext
            Get
                Return m_ctxFind
            End Get
            Set(value As PhoneBookContext)
                m_ctxFind = Value
            End Set
        End Property
        Private m_ctxFind As PhoneBookContext
        ' private field of type bool
        Private Property Find() As Boolean
            Get
                Return m_Find
            End Get
            Set(value As Boolean)
                m_Find = Value
            End Set
        End Property
        Private m_Find As Boolean
        ' private field of type string
        Private Property Charater() As String
            Get
                Return m_Charater
            End Get
            Set(value As String)
                m_Charater = Value
            End Set
        End Property
        Private m_Charater As String
        ' private field of type DataGridView
        Private Property Data() As DataGridView
            Get
                Return m_Data
            End Get
            Set(value As DataGridView)
                m_Data = Value
            End Set
        End Property
        Private m_Data As DataGridView
        ' private field of type string
        Private Property TypeSearch() As String
            Get
                Return m_TypeSearch
            End Get
            Set(value As String)
                m_TypeSearch = Value
            End Set
        End Property
        Private m_TypeSearch As String

        ' LinqToEntities using search function, which returns a DataGridView control containing the search results
        Public Function DetailPerson(ctx As PhoneBookContext, Find As Boolean, charater As String, search As String, data As DataGridView) As DataGridView
            ' Enhance the private variables with the parameters of the function DetailPerson
            Me.ctxFind = ctx
            Me.Find = Find
            Me.Charater = charater
            Me.Data = data
            Me.TypeSearch = search

            ' I declare a new instance of the class PhoneBookContext
            Using ctxFind As New PhoneBookContext()
                ' if the variable find and true charater and the variable is not empty or null
                If Me.Find.Equals(True) AndAlso Not String.IsNullOrEmpty(Me.Charater) Then
                    ' if the value of the variable equal to TypeSearch NAME 
                    If Me.TypeSearch.Equals("NAME") Then
                        ' I run the query LinqToEntities extrapolating the data starting with a certain letter in the variable specified in the entity name charater
						Dim query = From c In ctx.Persons Where c.NAME.StartsWith(Me.Charater)c

                        ' I make the iteration variable in the variable query person
                        For Each person As var In query
                            ' I post all the occurrences found in the variable of type DataGridView Date
                            Me.Data.Rows.Add(person.NAME, person.SURNAME, person.TELEPHONENUMBER, person.MOBILE)
                        Next
                    End If

                    ' if the value of the variable equal to TypeSearch SURNAME 
                    If Me.TypeSearch.Equals("SURNAME") Then
                        ' I run the query LinqToEntities extrapolating the data starting with a certain letter in the variable specified in the entity name charater
						Dim query = From c In ctx.Persons Where c.SURNAME.StartsWith(Me.Charater)c

                        ' I make the iteration variable in the variable query person
                        For Each person As var In query
                            ' I post all the occurrences found in the variable of type DataGridView Date
                            Me.Data.Rows.Add(person.NAME, person.SURNAME, person.TELEPHONENUMBER, person.MOBILE)
                        Next
                    End If

                    ' if the value of the variable equal to TypeSearch TELEPHONENUMBER 
                    If Me.TypeSearch.Equals("TELEPHONENUMBER") Then
                        ' I run the query LinqToEntities extrapolating the data starting with a certain letter in the variable specified in the entity name charater
						Dim query = From c In ctx.Persons Where c.TELEPHONENUMBER.StartsWith(Me.Charater)c

                        ' I make the iteration variable in the variable query person
                        For Each person As var In query
                            ' I post all the occurrences found in the variable of type DataGridView Date
                            Me.Data.Rows.Add(person.NAME, person.SURNAME, person.TELEPHONENUMBER, person.MOBILE)
                        Next
                    End If

                    ' if the value of the variable equal to TypeSearch MOBILE
                    If Me.TypeSearch.Equals("MOBILE") Then
                        ' I run the query LinqToEntities extrapolating the data starting with a certain letter in the variable specified in the entity name charater
						Dim query = From c In ctx.Persons Where c.MOBILE.StartsWith(Me.Charater)c

                        ' I make the iteration variable in the variable query person
                        For Each person As var In query
                            ' I post all the occurrences found in the variable of type DataGridView Date
                            Me.Data.Rows.Add(person.NAME, person.SURNAME, person.TELEPHONENUMBER, person.MOBILE)
                        Next
                    End If
                Else

                    ' In case of missing parameters
                    ' Display message to user
                    MessageBox.Show("Insert a Name,SurName,TelephoneNumber or Mobile for Person", Application.ProductName.ToString())
                End If
            End Using

            ' Returns the result of the function
            Return Me.Data
        End Function
    End Class
End Namespace

File: CS Practices/Example of Data Entry/C#/PhoneBook/FindPerson.cs
// Import the framework of the Libraries
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Data.Entity;

// PhoneBook project
namespace PhoneBook
{
    // class FindPerson
    class FindPerson
    {
        // private field of type PhooneBookContext
        private PhoneBookContext ctxFind { get; set; }
        // private field of type bool
        private bool Find {get;set;}
        // private field of type string
        private string Charater { get; set; }
        // private field of type DataGridView
        private DataGridView Data { get; set; }
        // private field of type string
        private string TypeSearch { get; set; }

        // LinqToEntities using search function, which returns a DataGridView control containing the search results
        public DataGridView DetailPerson(PhoneBookContext ctx,bool Find,string charater,string search,DataGridView data)
        {
            // Enhance the private variables with the parameters of the function DetailPerson
            this.ctxFind = ctx;
            this.Find = Find;
            this.Charater = charater;
            this.Data = data;
            this.TypeSearch = search;

            // I declare a new instance of the class PhoneBookContext
            using (PhoneBookContext ctxFind = new PhoneBookContext())
            {
                // if the variable find and true charater and the variable is not empty or null
                if (this.Find.Equals(true) &&  ! string.IsNullOrEmpty(this.Charater))
                {
                    // if the value of the variable equal to TypeSearch NAME 
                    if (this.TypeSearch.Equals("NAME"))
                    {
                        // I run the query LinqToEntities extrapolating the data starting with a certain letter in the variable specified in the entity name charater
                        var query = from c in ctx.Persons
                                    where c.NAME.StartsWith(this.Charater)
                                    select c;

                        // I make the iteration variable in the variable query person
                        foreach (var person in query)
                        {
                            // I post all the occurrences found in the variable of type DataGridView Date
                            this.Data.Rows.Add(person.NAME, person.SURNAME, person.TELEPHONENUMBER, person.MOBILE);
                        }
                    }

                    // if the value of the variable equal to TypeSearch SURNAME 
                    if (this.TypeSearch.Equals("SURNAME"))
                    {
                        // I run the query LinqToEntities extrapolating the data starting with a certain letter in the variable specified in the entity name charater
                        var query = from c in ctx.Persons
                                    where c.SURNAME.StartsWith(this.Charater)
                                    select c;

                        // I make the iteration variable in the variable query person
                        foreach (var person in query)
                        {
                            // I post all the occurrences found in the variable of type DataGridView Date
                            this.Data.Rows.Add(person.NAME, person.SURNAME, person.TELEPHONENUMBER, person.MOBILE);
                        }
                    }

                    // if the value of the variable equal to TypeSearch TELEPHONENUMBER 
                    if (this.TypeSearch.Equals("TELEPHONENUMBER"))
                    {
                        // I run the query LinqToEntities extrapolating the data starting with a certain letter in the variable specified in the entity name charater
                        var query = from c in ctx.Persons
                                    where c.TELEPHONENUMBER.StartsWith(this.Charater)
                                    select c;

                        // I make the iteration variable in the variable query person
                        foreach (var person in query)
                        {
                            // I post all the occurrences found in the variable of type DataGridView Date
                            this.Data.Rows.Add(person.NAME, person.SURNAME, person.TELEPHONENUMBER, person.MOBILE);
                        }
                    }

                    // if the value of the variable equal to TypeSearch MOBILE
                    if (this.TypeSearch.Equals("MOBILE"))
                    {
                        // I run the query LinqToEntities extrapolating the data starting with a certain letter in the variable specified in the entity name charater
                        var query = from c in ctx.Persons
                                    where c.MOBILE.StartsWith(this.Charater)
                                    select c;

                        // I make the iteration variable in the variable query person
                        foreach (var person in query)
                        {
                            // I post all the occurrences found in the variable of type DataGridView Date
                            this.Data.Rows.Add(person.NAME, person.SURNAME, person.TELEPHONENUMBER, person.MOBILE);
                        }
                    }
                }

                // In case of missing parameters
                else
                {
                    // Display message to user
                    MessageBox.Show("Insert a Name,SurName,TelephoneNumber or Mobile for Person",Application.ProductName.ToString());
                }
            }

            // Returns the result of the function
            return this.Data;
        }
    }
}
